// react-navigation-side-tabs/lib/module/views/HeaderViewNavigation.js
import React from 'react';
import { Text } from 'react-native';
import SideTabBar from './SideTabBar';
import * as T from '@react-navigation/stack';
import SafeAreaProviderCompat from './SafeAreaProviderCompat'; // color: rgb(28, 28, 30);
// font-size: 18px;
// font-weight: 500;

const HeaderView = ({
  headerTitle,
  tabBarLabel,
  descriptor
}) => {
  return /*#__PURE__*/React.createElement(SafeAreaProviderCompat, null, /*#__PURE__*/React.createElement(T.Header, {
    statusBarHeight: 60,
    insets: {
      top: 0
    },
    styleInterpolator: () => {
      return {
        titleStyle: {},
        leftButtonStyle: {},
        leftLabelStyle: {},
        rightButtonStyle: {},
        backgroundStyle: {}
      };
    },
    layout: {
      width: 0
    },
    scene: {
      progress: {
        current: true
      },
      route: {
        name: 'test2'
      },
      descriptor
    }
  }, /*#__PURE__*/React.createElement(Text, null, headerTitle || tabBarLabel)));
};

export default HeaderView;
