// react-navigation-side-tabs/lib/commonjs/views/Header/Header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.getDefaultHeaderHeight = void 0;

var _react = _interopRequireDefault(require("react"));

var _SideTabBar = _interopRequireDefault(require("../SideTabBar"));

var _reactNative = require("react-native");

var _native = require("@react-navigation/native");

var _reactNativeSafeAreaContext = require("react-native-safe-area-context");

var _HeaderTitle = _interopRequireDefault(require("./HeaderTitle"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const getDefaultHeaderHeight = statusBarHeight => {
  const layout = _reactNative.Dimensions.get('window');

  const isLandscape = layout.width > layout.height;
  let headerHeight;

  if (_reactNative.Platform.OS === 'ios') {
    // @ts-ignore
    if (isLandscape && !_reactNative.Platform.isPad) {
      headerHeight = 32;
    } else {
      headerHeight = 44;
    }
  } else if (_reactNative.Platform.OS === 'android') {
    headerHeight = 56;
  } else {
    headerHeight = 64;
  }

  return headerHeight + statusBarHeight;
};

exports.getDefaultHeaderHeight = getDefaultHeaderHeight;

const Header = ({
  descriptor,
  insets
}) => {
  const {
    colors
  } = (0, _native.useTheme)();
  const headerHeight = getDefaultHeaderHeight(insets.top);
  const {
    title,
    tabBarLabel,
    titleIcon,
    tabBarIcon
  } = descriptor.options;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, {
    pointerEvents: "box-none",
    style: [styles.header, {
      paddingTop: insets ? insets.top : 0,
      height: headerHeight,
      borderBottomColor: colors.border
    }]
  }, typeof titleIcon === 'function' && titleIcon(), /*#__PURE__*/_react.default.createElement(_HeaderTitle.default, {
    style: styles.text
  }, title || tabBarLabel));
};

const styles = _reactNative.StyleSheet.create({
  text: {
    marginLeft: 16
  },
  header: {
    elevation: 8,
    backgroundColor: '#FFF',
    paddingLeft: 16,
    alignItems: 'center',
    borderBottomWidth: _reactNative.StyleSheet.hairlineWidth,
    flexDirection: 'row'
  }
});

const HeaderWrapper = props => {
  return /*#__PURE__*/_react.default.createElement(_reactNativeSafeAreaContext.SafeAreaConsumer, null, insets => /*#__PURE__*/_react.default.createElement(Header, _extends({
    insets: insets
  }, props)));
};

var _default = HeaderWrapper;
exports.default = _default;
