// react-navigation-side-tabs/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "createSideTabNavigator", {
  enumerable: true,
  get: function get() {
    return _createSideTabNavigator.default;
  }
});
Object.defineProperty(exports, "SideTabView", {
  enumerable: true,
  get: function get() {
    return _SideTabView.default;
  }
});
Object.defineProperty(exports, "SideTabBar", {
  enumerable: true,
  get: function get() {
    return _SideTabBar.default;
  }
});
Object.defineProperty(exports, "ThemeProvider", {
  enumerable: true,
  get: function get() {
    return _theming.ThemeProvider;
  }
});
Object.defineProperty(exports, "withTheme", {
  enumerable: true,
  get: function get() {
    return _theming.withTheme;
  }
});
Object.defineProperty(exports, "useTheme", {
  enumerable: true,
  get: function get() {
    return _theming.useTheme;
  }
});

var _createSideTabNavigator = _interopRequireDefault(require("./navigators/createSideTabNavigator"));

var _SideTabView = _interopRequireDefault(require("./views/SideTabView"));

var _SideTabBar = _interopRequireDefault(require("./views/SideTabBar"));

var _theming = require("./theming");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
